contracts/FungibleTokenSwitchboard.cdc

```diff
@@ -0,0 +1,143 @@
+            // Emit the event that indicates that a new capability has been added
+            emit VaultCapabilityAdded(type: vaultRef.getType())
+        }
+
+        /// removeVault removes a fungible token receiver capability 
+        ///                       from the switchboard resource
+        /// 
+        /// Parameters: capability: The capability to a fungible token vault 
+        ///                         to be removed from the switchboard
+        ///
+        pub fun removeVault(capability: Capability<&{FungibleToken.Receiver}>) {
+            // Borrow a reference to the vault pointed by the capability we want
+            // store inside the switchboard            
+            let vaultRef = capability.borrow() 
+                ?? panic ("Cannot borrow reference to vault from capability")
+            // Use the vault reference to find the capability to remove
+            self.receiverCapabilities.remove(key: vaultRef.getType())
+            // Emit the event that indicates that a new capability has been removed
+            emit VaultCapabilityRemoved(type: vaultRef.getType())            
+        }
+        
+        /// deposit Takes a fungible token vault and routes it to the proper
+        ///         fungible token receiver capability for depositing it
+        /// 
+        /// Parameters: from: The deposited fungible token vault resource
+        ///        
+        pub fun deposit(from: @FungibleToken.Vault) {
+            let depositedVaultCapability = self.receiverCapabilities[from.getType()] ?? 
+                panic ("The deposited vault is not available on this switchboard")
+            let vaultRef = depositedVaultCapability.borrow() ?? 
+                panic ("Can not borrow a reference to the the vault")
+            vaultRef.deposit(from: <- from)
+        }
+
+        /// getVaultCabilities function for get to know which tokens a certain
+        /// switchboard resource is prepared to receive
+        ///
+        /// Returns: The dictionary of stored {FungibleToken.Receiver} 
+        /// capabilities
+        ///
+        pub fun getVaultCapabilities(): {Type: Capability<&{FungibleToken.Receiver}>} {
+            return self.receiverCapabilities
+        }
+
+        init() {
+            // Initialize the capabilities dictionary
+            self.receiverCapabilities = {}
+            // Emit the event that indicates that a new switchboard resource
+            // has been created
+            emit SwitchboardInitialized(switchboardResourceID: self.uuid)
+        }
+    }
+
+    /// createSwitchboard
+    ///
+    /// Function that allows to create a new blank switchboard. A user must call
+    /// this function and store the returned resource in their storage
+    ///
+    pub fun createSwitchboard(): @Switchboard {
+        return <- create Switchboard()
+    }
+
+    init() {
+        self.StoragePath = StoragePath(identifier: "fungibleTokenSwitchboard")!
```

Comment:
nitpick little but  isn't it better to use non-dynamic paths in this context ?

`self.StoragePath = /storage/fungibleTokenSwitchboard` etc

Reply:
Honestly I just knew about the dynamic function for creating paths while coding this and wanted to try it, do you know if there's any specific advantages / disadvantages  for each way?

Reply:
as I said there is no disadvantage, just syntax nitpick.